File: test_api.js
const fs = require('fs');
const https = require('https');

// We need a tiny valid base64 image (a 1x1 red pixel) to send to the server
const redPixel = "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z8BQDwAEhQGAhKmMIQAAAABJRU5ErkJggg==";

const data = JSON.stringify({
  imageBase64: `data:image/png;base64,${redPixel}`,
  mimeType: "image/png",
  patientName: "Test Patient",
  age: "30",
  gender: "Masculino",
  additionalNotes: "Prueba técnica"
});

const options = {
  hostname: 'aivitaleye-production.up.railway.app',
  port: 443,
  path: '/api/analyze-photo',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': data.length
  }
};

const req = https.request(options, (res) => {
  let responseData = '';

  res.on('data', (chunk) => {
    responseData += chunk;
  });

  res.on('end', () => {
    console.log('Status Code:', res.statusCode);
    console.log('Response:', responseData);
  });
});

req.on('error', (error) => {
  console.error('Error:', error);
});

req.write(data);
req.end();

File: test_api.py
import urllib.request
import json

red_pixel = "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z8BQDwAEhQGAhKmMIQAAAABJRU5ErkJggg=="
data = {
    "imageBase64": f"data:image/png;base64,{red_pixel}",
    "mimeType": "image/png",
    "patientName": "Test Patient",
    "age": "30",
    "gender": "Masculino",
    "additionalNotes": "Prueba técnica"
}

req = urllib.request.Request(
    "https://aivitaleye-production.up.railway.app/api/analyze-photo",
    data=json.dumps(data).encode('utf-8'),
    headers={"Content-Type": "application/json"}
)

try:
    with urllib.request.urlopen(req) as response:
        print("Status Code:", response.status)
        print("Response:", response.read().decode('utf-8'))
except urllib.error.HTTPError as e:
    print("HTTP Error:", e.code)
    print(e.read().decode('utf-8'))
except Exception as e:
    print("Error:", e)
